model/jpa/src/main/java/org/keycloak/models/jpa/entities/UserEntity.java

```diff
@@ -50,7 +50,7 @@
         @NamedQuery(name="getRealmUserCount", query="select count(u) from UserEntity u where u.realmId = :realmId"),
         @NamedQuery(name="getRealmUserCountExcludeServiceAccount", query="select count(u) from UserEntity u where u.realmId = :realmId and (u.serviceAccountClientLink is null)"),
         @NamedQuery(name="getRealmUsersByAttributeNameAndValue", query="select u from UserEntity u join u.attributes attr " +
-                "where u.realmId = :realmId and attr.name = :name and attr.value = :value"),
+                "where u.realmId = :realmId and attr.name = :name and lower(attr.value) = :value"),
```

Comment:
Aren't we changing the expectation around the default collation used by the database? As we are basically querying custom attributes, I might not want `foo` the same as `FoO`. And that is how we have been working.